node_package/src/serverRenderReactComponent.ts

```diff
@@ -36,16 +39,27 @@ function processServerRenderHash(result: ServerRenderResult, options: RenderOpti
     // Possibly, someday, we could have the Rails server redirect.
     htmlResult = '';
   } else {
-    htmlResult = result.renderedHtml as string;
+    htmlResult = result.renderedHtml;
   }
 
-  return { result: htmlResult, hasErrors };
+  return { result: htmlResult ?? null, hasErrors };
+}
+
+function processReactElement(result: ReactElement): string {
+  try {
+    return ReactDOMServer.renderToString(result);
+  } catch (error) {
+    console.error(`Invalid call to renderToString. Possibly you have a renderFunction, a function that already
+calls renderToString, that takes one parameter. You need to add an extra unused parameter to identify this function
+as a renderFunction and not a simple React Function Component.`);
+    throw error;
+  }
 }
 
 function processPromise(
-  result: Promise<string>,
+  result: CreateReactOutputAsyncResult,
   renderingReturnsPromises: boolean,
-): Promise<string> | string {
+): RenderStateHtml {
   if (!renderingReturnsPromises) {
     console.error(
       'Your render function returned a Promise, which is only supported by a node renderer, not ExecJS.',
```

Comment:
I realize the PR doesn't touch it, but maybe
```suggestion
      'Your render function returned a Promise, which is only supported by the React on Rails Pro Node renderer, not ExecJS.',
```
?